### DISCUSSION TITLE
Looking for a reliable zigbee router

### DISCUSSION THREAD

**Discussion by isabido** [Koenkk/zigbee2mqtt#13504] [Devices]
HI friends, I am looking for a reliable router to expand the range of my SONOFF Zigbee 3.0 USB Dongle.

In view of the problems many sonoff devices such as the zbmini and s26 sockets are having, I have no idea which way to go.

Will it be the most reliable Philips HUE plugs as routers? https://www.zigbee2mqtt.io/devices/929002240401.html

Or maybe other SONOFF Zigbee 3.0 USB Dongle in router mode?

Thanks in advance!

**Comment by zen2** [comment]
I manage a mesh network actual composed of ~100 devices with 39 routers.
By experience all mains powered devices act as routers but some are really better in the purpose to expand a network.

To expand a network I advice these cheap devices:
![image](https://user-images.githubusercontent.com/9433239/184320872-41285fb8-c28f-4956-bc44-85875ad05ac1.png)

You can find them easily under different brand using search keyword "zigbee on/off controller" even on mamazon for a price ~10$.

**Reply by isabido** [reply]
Thanks for the recommendation, indeed I have several Philips HUE lights that are acting as a router, but I will have to remove them because in the end they turn off from the switch and there is chaos in the zigbee network. I bought several ZBMINI-Ls to change them and finally these do not function as Routers as they work without NEUTRAL.

In short, I do not have reliable routers in the network, and I was looking for devices that really work well.

By the way, do we know inside this "zigbee on/off controller" which chipset they use?

**Reply by zen2** [reply]
I don't know which chipset it is using. For me they are simply reliable and cheap.
I got two of them in strategic place and they are connected respectively with 24 and 17 devices in the mesh.
After you can use several powermeter plugs too I presume. I use a lot of them.
I prefer these cheap devices because since you have to wire them you know that they won't be disconnect and have impact on your mesh.
I use one above kitchen sink to control light from a led tube that is activated by a presence sensor next to it.
The second one is simply plug at second level floor and control nothing actually but I've added it to get a better mesh at second floor since I get too many end device disconnections through time even if the whole floor use HUE bulbs that act as repeter. Since I do not lost any more end devices.

**Comment by lux73** [comment]
I am using this dedicated Stück with Router-std Firmware -> https://www.ebay.de/itm/274300792162?hash=item3fdd99ed62:g:cRcAAOSw3MRiJjS4

Works without any hiccup

**Comment by zen2** [comment]
Ho I forget to mention, these Zigbee On/Off controller are recognised by zigbee2mqtt as:
https://www.zigbee2mqtt.io/devices/SA-003-Zigbee.html

**Comment by jerrm** [comment]
I have a couple Sonoff Dongle+ units flashed with router firmware tucked away in out of sight spots.   Pretty much all the endpoints have migrated to route through them. Get a charger where the usb port is at right angle to the plug so that the dongle hugs the wall. 

Previously, a mix of Sengled and Sonoff smart plugs served the function. 

I have a couple always powered zigbee bulbs, but have stayed away from any switched zigbee bulbs for exactly your reasons.  Most of my lighting is on lutron smart switches. I do have a few switched wifi smart bulbs where I want color, or seldom used locations that don't justify a lutron smart switch and where I want to make sure the light will turn itself off if the manual switch is left on.

**Reply by isabido** [reply]
Jerm, I've read comments that the dongle + stays "out" after a few days or hours of being running. This is not your experience right?

**Reply by jerrm** [reply]
The first one I deployed as a router had no issues at all, but it almost immediately had a dozen devices attached.

The second one  in a more remote location only had a couple of endpoints attached at first.  It would go unavailable after some time.  The fix was to set min/max intervals under the reporting tab for the device.   I have had absolutely no probleme since setting the reporting interval (several months now).  At this point about a third of my network attaches to the device, so any problems would be noticed quickly.

I'm assuming, but never really tested enough to confirm, the firmware on the relatively "quiet" stick didn't ping home often enough by default.

![image](https://user-images.githubusercontent.com/12073505/184501773-c7c266a0-64e5-4c07-927c-13e586986ae3.png)

**Reply by isabido** [reply]
Thanks for such a detailed answer. I still have to delve into how to diagnose a zigbee network, I'm a novice with these connections.

At the moment I see this type of network, in the mesh routings when I turn off a router (Philips HUE lamps) all the end nodes take a long time to realize and change their connection.

![image](https://user-images.githubusercontent.com/559650/184505740-6e62f9f7-7486-4c8f-9427-3a3dec4cc7df.png)

**Reply by jerrm** [reply]
Odds are pretty good most devices would choose the Sonoff over the bulbs if they are both at similar distances, but there isn't any guarantee.

If the bulb is switched, I use a wifi option to avoid the re-route problems.

**Comment by isabido** [comment]
Thank you all for writing your experience. I think I'm finally going to decide to install two more Sonoff Dongle+ units flashed with router firmware, so it will be the same as the coordinator.

**Reply by ut1545** [reply]
Hello isabido,
My setup is based on Sonoff Dongle plus, as coordinator and 2 routers. Works perfectly.
Also, in your map we can see a endpoint (puerta entrada) with two doted links. What this means? Until now, I always was thinking that a endpoint could have only one link to a router or coordinator.... I'm confused.

**Reply by isabido** [reply]
thanks for your answer! I will definitely go for the full solution with dongle plus. The truth is that I don't know why there is that device with two dotted lines. I think that being passive and not opening the door, it had not sent information to the router/coordinator after the last reboot.
